presto-native-sidecar-plugin/src/main/java/com/facebook/presto/sidecar/sessionpropertyproviders/NativeSystemSessionPropertyProvider.java

```diff
@@ -0,0 +1,232 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.sidecar.sessionpropertyproviders;
+
+import com.facebook.airlift.json.JsonCodec;
+import com.facebook.airlift.log.Logger;
+import com.facebook.presto.common.type.Type;
+import com.facebook.presto.common.type.TypeManager;
+import com.facebook.presto.spi.NodeManager;
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.session.PropertyMetadata;
+import com.facebook.presto.spi.session.SessionPropertyMetadata;
+import com.facebook.presto.spi.session.SystemSessionPropertyProvider;
+import com.google.inject.Inject;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.facebook.presto.common.type.BigintType.BIGINT;
+import static com.facebook.presto.common.type.BooleanType.BOOLEAN;
+import static com.facebook.presto.common.type.DoubleType.DOUBLE;
+import static com.facebook.presto.common.type.IntegerType.INTEGER;
+import static com.facebook.presto.common.type.TinyintType.TINYINT;
+import static com.facebook.presto.common.type.VarcharType.VARCHAR;
+import static com.facebook.presto.spi.StandardErrorCode.INVALID_ARGUMENTS;
+import static com.facebook.presto.spi.StandardErrorCode.INVALID_SESSION_PROPERTY;
+import static com.facebook.presto.spi.session.PropertyMetadata.booleanProperty;
+import static com.facebook.presto.spi.session.PropertyMetadata.doubleProperty;
+import static com.facebook.presto.spi.session.PropertyMetadata.integerProperty;
+import static com.facebook.presto.spi.session.PropertyMetadata.longProperty;
+import static com.facebook.presto.spi.session.PropertyMetadata.stringProperty;
+import static com.facebook.presto.spi.session.PropertyMetadata.tinyIntProperty;
+import static java.util.Objects.requireNonNull;
+
+public class NativeSystemSessionPropertyProvider
+        implements SystemSessionPropertyProvider
+{
+    private static final Logger log = Logger.get(NativeSystemSessionPropertyProvider.class);
+    private final NodeManager nodeManager;
+    private final TypeManager typeManager;
+    private final JsonCodec<List<SessionPropertyMetadata>> nativeSessionPropertiesJsonCodec;
+
+    @Inject
+    public NativeSystemSessionPropertyProvider(
+            JsonCodec<List<SessionPropertyMetadata>> nativeSessionPropertiesJsonCodec,
+            NodeManager nodeManager, TypeManager typeManager)
+    {
+        this.nativeSessionPropertiesJsonCodec = nativeSessionPropertiesJsonCodec;
+        this.nodeManager = nodeManager;
+        this.typeManager = typeManager;
+    }
+
+    @Override
+    public List<PropertyMetadata<?>> getSessionProperties()
+    {
+        return fetchSessionProperties();
+    }
+
+    private List<PropertyMetadata<?>> fetchSessionProperties()
```

Comment:
Instead of throwing dummy values, let's just throw, since we add this functionality in the subsequent commit.